var/spack/repos/builtin/packages/py-jaxlib/package.py

```diff
@@ -99,11 +99,12 @@ class PyJaxlib(PythonPackage, CudaPackage, ROCmPackage):
         depends_on("py-build", when="@0.4.14:")
 
     with default_args(type=("build", "run")):
+        depends_on("python@3.10:", when="@0.4.31:")
+        depends_on("python@3.9:", when="@0.4.14:")
+        depends_on("python@3.8:", when="@0.4.6:")
         # Based on PyPI wheels
-        depends_on("python@3.10:3.12", when="@0.4.31:")
-        depends_on("python@3.9:3.12", when="@0.4.17:0.4.30")
-        depends_on("python@3.9:3.11", when="@0.4.14:0.4.16")
-        depends_on("python@3.8:3.11", when="@0.4.6:0.4.13")
```

Comment:
Can you add:
```python
depends_on("python@:3.13")
```
Someday we will have to add 3.14.

Reply:
imo that leads to more work in the future, removing redundant bounds. I'd rather add upperbounds if things are known to break.

added this one though.